runtime/parser/lexer/state.go

```diff
@@ -118,6 +129,17 @@ func rootState(l *lexer) stateFn {
 			return numberState
 		case '"':
 			return stringState
+		case '\\':
+			if l.mode == STR_IDENTIFIER {
+				r = l.next()
+				switch r {
+				case '(':
+					l.emitType(TokenStringTemplate)
+					l.openBrackets++
```

Comment:
So after reaching `\(`, it continues to lex in the root-state. So AFAIU, lexer doesn't limit what can be put inside the interpolation (only the parser does, correct?). Curious how does the lexer handles nested string templates? Something like `"outer template \( "inner template with \(foo)") end"`. Could you maybe add some tests?

Reply:
Tests added in [bda6b28](https://github.com/onflow/cadence/pull/3585/commits/bda6b28fb29cff50e635a3e9d94b926290001b33). For `"outer template \( "inner template with \(foo)") end"`, the parser will identify `"outer template \( "` as a string which is invalid and will error with `missing ')'`.